src/restapi/mm_routes.ts

```diff
@@ -0,0 +1,49 @@
+import fs from 'fs';
+
+import express, {Request, Response} from 'express';
+import {AppCall} from 'mattermost-redux/types/apps';
+
+import app from '../app/app';
+
+import {jsonStoreFileName} from '../app/constants';
+
+const router = express.Router();
+
+router.post('/createform', async (req: Request, res: Response) => {
+    const appCall: AppCall = req.body;
+    if (appCall.type === 'form') {
+        const createForm = app.calls.createForm(appCall.context.post.message);
+        res.json(createForm);
+    } else {
+        const message = app.createTicketFromPost(appCall);
```

Comment:
Why assign the return value to this variable and then not use it anywhere? Also we should use `await` here so we can catch an error if it occurs.

Reply:
added the await and also catch any errors within this function call and display in the modal on submit.
https://github.com/mattermost/mattermost-app-zendesk/commit/09732405a9469420730de8af14c8a1ba9d1164e1

Reply:
I'm thinking we should have an uncaught error handler for all routes (in case our try/catches don't catch the error). We can return a call response with a generic error message. Separate ticket for this is good